show-labels-on-layer/src/main/java/com/esri/arcgismaps/sample/showlabelsonlayer/MainActivity.kt

```diff
@@ -0,0 +1,146 @@
+        // US Congressional Districts
+        val serviceFeatureTable =
+            ServiceFeatureTable(getString(R.string.congressional_districts_url))
+        // create the feature layer from the service feature table
+        val featureLayer = FeatureLayer(serviceFeatureTable)
+        // add this feature layer to the map
+        map.operationalLayers.add(featureLayer)
+        // add the map to the mapview
+        mapView.map = map
+
+        lifecycleScope.launch {
+            // if the map load failed show an error and return
+            map.load().onFailure {
+                return@onFailure showError("Error loading map:${it.message}")
+            }
+            // if the feature layer load failed show an error and return
+            featureLayer.load().onFailure {
+                return@onFailure showError("Error loading feature layer:${it.message}")
+            }
+            // zoom to the layer when it's done loading
+            featureLayer.fullExtent?.let { extent ->
+                mapView.setViewpoint(Viewpoint(extent))
+            }
+        }
+
+        // create label definitions for the different party type attribute in the feature service
+        val republicanLabelDefinition = createLabelDefinition("Republican", Color.red)
+        val democraticLabelDefinition = createLabelDefinition("Democrat", Color.blue)
+
+        featureLayer.apply {
+            // add the label definitions to the feature layer
+            labelDefinitions.addAll(
+                listOf(
+                    republicanLabelDefinition,
+                    democraticLabelDefinition
+                )
+            )
+            // enable the labels
+            featureLayer.labelsEnabled = true
+        }
+    }
+
+    /**
+     * Creates and returns a [LabelDefinition] for the given party [name] attribute
+     * with the [labelColor]
+     */
+    private fun createLabelDefinition(
+        name: String,
+        labelColor: Color
+    ): LabelDefinition {
+        // create a text symbol for styling the label
+        val textSymbol = TextSymbol().apply {
+            color = labelColor
+            size = 12f
+            haloColor = Color.white
+            haloWidth = 2f
+        }
+        // create a arcade label expression for the label text
+        val arcadeLabelExpression =
+            ArcadeLabelExpression(
+                "\$feature.NAME + \" (\" + left(\$feature.PARTY,1) " +
+                    "+ \")\\nDistrict \" + \$feature.CDFIPS"
+            )
+        // create and return a new label definition with the arcadeLabelExpression and textSymbol
+        return LabelDefinition(arcadeLabelExpression, textSymbol).apply {
+            // set the label placement
+            placement = LabelingPlacement.PolygonAlwaysHorizontal
+            // set the attribute name for which this label will be generated
+            whereClause = String.format("PARTY = '%s'", name)
```

Comment:
I think this can be done with Kotlin string interpolation instead
```suggestion
            whereClause = "PARTY = $name"
```

Reply:
This is how I had it initially but the labels wouldn't show up. After some testing it turned out it only works when `String.format` is used. Not sure why..

Reply:
Ah that did it, thanks!